library/src/main/java/com/chuckerteam/chucker/internal/ui/MainViewModel.kt

```diff
@@ -35,6 +42,22 @@ internal class MainViewModel : ViewModel() {
     val throwables: LiveData<List<RecordedThrowableTuple>> = RepositoryProvider.throwable()
         .getSortedThrowablesTuples()
 
+    fun getAllTransactions(): List<HttpTransaction>? = runBlocking {
```

Comment:
With `runBlocking` you just killing the whole point of using coroutines. No matter what you switched to other dispatcher below - you already blocked the main thread till this code finishes.

Reply:
I am kinda new to the coroutines concept, thanks for pointing it out.

so I tried a couple of approaches here
1. passing a function as a callback argument, but as you mentioned it seems to make the function hard to read.
2. keep Livedata list of httptransaction and update it after your fetch, but it seemed overkill to observe for exporting transactions.

Hence the final approach for which I settled here was adding `lifecycle-runtime-ktx` so that I would be able to launch coroutine in lifecycle scope, and it is canceled when the Lifecycle is destroyed. 

Let me know if you have any alternate solution or think adding `lifecycle-runtime-ktx` dependency is unnecessary.

Reply:
I am not a fan of adding more dependencies. Take a look at `TransactionPayloadFragment` to see how another contributor did pretty similar stuff.

Reply:
cool thanks for the pointer, updated.